src/autopas/containers/verletListsCellBased/verletListsCells/VerletListsCells.h

```diff
@@ -115,12 +115,183 @@ class VerletListsCells : public VerletListsLinkedBase<Particle> {
+          return particlePtr == &p1;
+        });
+        // If yes, append p2 to it.
+        if (iter != neighborList.end()) {
+          iter->second.push_back(&p2);
+        } else {
+          // If no, create one (or reuse an empty pair), reserve space for the list and emplace p2
+          if (auto insertLoc = std::find_if(neighborList.begin(), neighborList.end(),
+                                            [&](const auto &pair) {
+                                              const auto &[particlePtr, list] = pair;
+                                              return particlePtr == nullptr;
+                                            });
+              insertLoc != neighborList.end()) {
+            auto &[particlePtr, neighbors] = *insertLoc;
+            particlePtr = &p1;
+            neighbors.reserve(listLengthEstimate);
+            neighbors.push_back(&p2);
+          } else {
+            neighborList.emplace_back(&p1, std::vector<Particle *>{});
+            neighborList.back().second.reserve(listLengthEstimate);
+            neighborList.back().second.push_back(&p2);
+          }
+        }
+      }
+    };
+
+    // Vector of offsets from the base cell for the c08 base step
+    // and respective factors for the fraction of particles per cell that need neighbor lists in the base cell.
+    const auto offsetsC08 =
+        VerletListsCellsHelpers::buildC08BaseStep(utils::ArrayUtils::static_cast_copy_array<int>(cellsPerDim));
+
+    // Go over all cells except the very last layer and create lists per base step.
+    // Since there are no loop dependencies merge all for loops and create 10 chunks per thread.
+    AUTOPAS_OPENMP(parallel for collapse(3) schedule(dynamic, std::max(cells.size() / (autopas::autopas_get_max_threads() * 10), 1ul)))
+    for (int z = 0; z < cellsPerDim[2] - 1; ++z) {
+      for (int y = 0; y < cellsPerDim[1] - 1; ++y) {
+        for (int x = 0; x < cellsPerDim[0] - 1; ++x) {
+          // aliases
+          const auto cellIndexBase = utils::ThreeDimensionalMapping::threeToOneD(x, y, z, cellsPerDim);
+          auto &baseCell = cells[cellIndexBase];
+          auto &baseCellsLists = neighborLists[cellIndexBase];
+
+          // Allocate space for ptr-list pairs for this cell
+          baseCellsLists.resize(
+              VerletListsCellsHelpers::estimateNumLists(cellIndexBase, this->_verletBuiltNewton3, cells, offsetsC08));
+          for (size_t i = 0; i < baseCell.size(); ++i) {
+            // if there is already an unused ptr-list pair use it, else emplace a new one.
+            if (i < baseCellsLists.size()) {
+              auto &[particlePtr, neighbors] = baseCellsLists[i];
+              particlePtr = &baseCell[i];
+              neighbors.reserve(listLengthEstimate);
+            } else {
+              baseCellsLists.emplace_back(&baseCell[i], std::vector<Particle *>{});
+              baseCellsLists.back().second.reserve(listLengthEstimate);
+            }
```

Comment:
Here, two things happen:
1. The neighbor lists are initialized. This can mean one of two things:
  1. (line 217/218) A neighbor lists that was previously cleared and marked as unused (first member of the pair, the particle pointer to which the neighbor list belongs, was set to `nullptr`) is reactivated.
  2. (line 221) A new neighbor list is initialized.
2. The actual neighbor lists (lists of particle pointers) are pre-allocated. This is to avoid any reallocations when they are populated by `push_back` operations.

Added a mini comment -> 8ee84c328 
Unresolve this issue if you feel this is not enough.